src/pipeline/rackhd/docker/publish.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash -e
+# Environmental requirement:
+# docker service running and docker have already logged with Rackhd Dockerhub ID,
+#     cmd 'docker login', if not logged then can't push images to dockerhub
+
+#########################################
+#
+#  Usage
+#
+#########################################
+Usage(){
+    echo "function: this script is used to push rackhd docker images to docker hub"
+    echo "usage: $0 [options] [arguments]"
+    echo "    mandatory arguments:"
+    echo "      --DOCKER_STASH_PATH: The directory path of docker images tar file."
+    echo "      --DOCKERHUB_USER: the username of dockerhub"
+    echo "      --DOCKERHUB_PASS: Password for DOCKERHUB_USER of dockerhub."
+}
+
+###################################################################
+#
+#  Parse and check Arguments
+#
+##################################################################
+parseArguments(){
+    while [ "$1" != "" ]; do
+        case $1 in
+            --DOCKER_STASH_PATH )           shift
+                                            DOCKER_STASH_PATH=$1
+                                            ;;
+            --DOCKERHUB_USER )              shift
+                                            DOCKERHUB_USER=$1
+                                            ;;
+            --DOCKERHUB_PASS )              shift
+                                            DOCKERHUB_PASS=$1
+                                            ;;
+           * )
+                                            Usage
+                                            exit 1
+        esac
+        shift
+    done
+    if [ ! -n "${DOCKER_STASH_PATH}" ]; then
+        echo "[Error]Arguments DOCKER_STASH_PATH is required"
+        exit 1
+    fi
+    if [ ! -n "${DOCKERHUB_USER}" ]; then
+        echo "[Error]Arguments DOCKERHUB_USER is required"
+        exit 1
+    fi
+    if [ ! -n "${DOCKERHUB_PASS}" ]; then
+        echo "[Error]Arguments DOCKERHUB_PASS is required"
+        exit 1
+    fi
+}
+
+# Clean UP. (was in Jenkins job post-build, avoid failure impacts build status.)
+cleanUp(){
+    # parameter : the images keyword to be delete
+    keyword=$1
+    images=`docker images | grep ${keyword} | awk '{print $3}' | sort | uniq`
+    if [ -n "$images" ]; then
+        docker rmi -f $images
+    fi
+}
+
+cleanRunningContainers() {
+    local containers=$(docker ps -a -q)
+    if [ "$containers" != "" ]; then
+        echo "Clean Up containers : " ${containers}
+        docker stop ${containers}
+        docker rm  ${containers}
+    fi
+}
+
+########################################################
+#
+# Main
+#
+########################################################
+
+main(){
```

Comment:
I recommand to cut this main function into serval small functions, for example:
```
loadImages
push
clean
```
one method one function is easy to maintain, especially for next maintainers.

Reply:
Good idea.